### PR TITLE
added moments in code and in ps1

### PR DESCRIPTION
**Pull request by GustaveHerve**

### CODE REVIEW

File: Holovibes/cuda_includes/tools_compute.cuh [hunk 1]
```diff
@@ -56,3 +56,14 @@ void gpu_normalize(float* const input,
...
+/*! \brief Performs tensor vector multiplication. Parallelisation is done along frame_res.
+ * \param output the place to sae
+*/
+void tensor_multiply_vector(float* output,
+                            const float* tensor,
+                            const float* vector,
+                            const uint frame_res,
+                            const ushort f_start,
+                            const ushort f_end,
+                            const cudaStream_t stream);
```
Comment (noTban) [inline Holovibes/cuda_includes/tools_compute.cuh:69]: Doc missing

File: Holovibes/sources/cli.cc [hunk 2]
```diff
@@ -235,7 +251,7 @@ static void main_loop(holovibes::Holovibes& holovibes)
             else
             {
                 // Change the speed of the progress bar according to the nb of frames skip
-                progress_bar(progress->first * (holovibes::api::get_nb_frame_skip() + 1), progress->second, 40);
+                progress_bar(progress->first, progress->second, 40);
```
Comment (Baschtien) [inline Holovibes/sources/cli.cc:254]: Merge conflict ?

File: Holovibes/sources/compute/fourier_transform.cc [hunk 3]
```diff
@@ -235,6 +235,92 @@ void FourierTransform::insert_stft()
...
+            if (type == ImgType::Moments_2)
+            {
+                tensor_multiply_vector(buffers_.gpu_postprocess_frame,
+                                       moments_env_.stft_res_buffer,
+                                       moments_env_.f2_buffer,
+                                       fd_.get_frame_res(),
+                                       moments_env_.f_start,
+                                       moments_env_.f_end,
+                                       stream_);
+            }
```
Comment (Baschtien) [inline Holovibes/sources/compute/fourier_transform.cc:320]: Maybe compact all this, as only the moments_env_.fx buffer changes

File: Holovibes/sources/gui/windows/MainWindow.cc [hunk 4]
```diff
@@ -765,6 +765,7 @@ void MainWindow::set_composite_values()
 
 void MainWindow::set_view_image_type(const QString& value)
 {
+    // TODO Moments come here when select in the middle raw
```
Comment (Baschtien) [inline Holovibes/sources/gui/windows/MainWindow.cc:768]: Remove / fix TODO

File: Holovibes/sources/gui/windows/panels/export_panel.cc [hunk 5]
```diff
@@ -203,6 +205,8 @@ void ExportPanel::set_record_mode(const QString& value)
     stop_record();
 
     const std::string text = value.toStdString();
+    LOG_WARN("Record mode chosen :" + text);
+    // TODO: configure for Moments
```
Comment (Baschtien) [inline Holovibes/sources/gui/windows/panels/export_panel.cc:209]: Maybe remove ?

**Review by Baschtien — CHANGES_REQUESTED** [review]

File: Holovibes/sources/cli.cc (hunk 2 above)
Comment (Hpn4) [reply Holovibes/sources/cli.cc:254]: The calculation inside the `cli` for the progress bar recording where wrong. The frame skip is already calculated inside the `RecordWorker` so you just need to get how many frames were recorded

File: Holovibes/includes/compute/matrix_operations.hxx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+void matrix_multiply(const T* A,
+                     const T* B,
+                     int A_height,
+                     int B_width,
+                     int A_width_B_height,
+                     T* C,
+                     cublasOperation_t op_A,
+                     cublasOperation_t op_B)
+{
+    // LOG-USELESS LOG_FUNC();
```
Comment (TitouanGragnic) [inline Holovibes/includes/compute/matrix_operations.hxx:21]: deadcode?

**Review by Baschtien — APPROVED** [review]

File: Holovibes/includes/enum/enum_img_type.hh [hunk 7]
```diff
@@ -19,7 +18,10 @@ enum class ImgType
     SquaredModulus, /*!<  Modulus taken to its square value */
     Argument,       /*!<  Phase (angle) value of the complex pixel c, computed with atan(Im(c)/Re(c)) */
     PhaseIncrease,  /*!<  Phase value, the conjugate between the phase of the last image and the previous one */
-    Composite       /*!<  Displays different frequency intervals on color RBG or HSV chanels*/
+    Composite,      /*!<  Displays different frequency intervals on color RBG or HSV chanels*/
+    Moments_0,      /*!<  Displays the moments at order 0, 1 and 2. NOT IMPLEMENTED YET*/
```
Comment (TitouanGragnic) [inline Holovibes/includes/enum/enum_img_type.hh:22]: Implemented or not implemented?

File: Holovibes/sources/compute/fourier_transform.cc [hunk 8]
```diff
@@ -235,6 +235,92 @@ void FourierTransform::insert_stft()
...
+
+            if (type == ImgType::Moments_0)
+            {
+                /*size_t index = fd_.get_frame_res() * (moments_env_.f_start);
+                cudaXMemcpyAsync(buffers_.gpu_postprocess_frame,
+                                 moments_env_.stft_res_buffer + index,
+                                 fd_.get_frame_res() * sizeof(float),
+                                 cudaMemcpyDeviceToDevice,
+                                 stream_);*/
```
Comment (TitouanGragnic) [inline Holovibes/sources/compute/fourier_transform.cc]: deadcode?

File: Holovibes/sources/thread/frame_record_worker.cc [hunk 9]
```diff
@@ -222,12 +221,8 @@ void FrameRecordWorker::run()
...
 void FrameRecordWorker::wait_for_frames()
 {
-    auto pipe = Holovibes::instance().get_compute_pipe();
-    while (!stop_requested_)
-    {
-        if (record_queue_.load()->get_size() != 0)
-            break;
-    }
+    while (!stop_requested_ && record_queue_.load()->get_size() == 0)
+        continue;
```
Comment (TitouanGragnic) [inline Holovibes/sources/thread/frame_record_worker.cc:225]: try if it's work in release mode, maybe remove by optimisation

**Review by TitouanGragnic — CHANGES_REQUESTED** [review]

**Review by TitouanGragnic — APPROVED** [review]

**Review by ArthurCourselle — APPROVED** [review]

**Pull request merged**